src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -247,8 +249,11 @@ def update_argument(self, param_name, argtype):
         arg.type.update(other=argtype)
 
     def _resolve_default_value_from_cfg_file(self, arg, overrides):
+        if arg.type.required == None:
+            arg.type.required = arg.type.settings.get('required', False)
```

Comment:
Could you please elaborate how those extra fields to be used later? Are you letting CLI dump out a command table file and analyze later?